### PR TITLE
refactor: turn ToggleWrapper into a function component

### PR DESCRIPTION
**Pull request by leonardomaier**
# Description

Refactor ToggleWrapper from a class component to a function component

## Which issue(s) does this PR relate to?

https://github.com/dcos-labs/ui-kit/issues/796

## Testing

Testing can be done via storybook and by running unit tests.

## Trade-offs

None

## Screenshots

Not needed.

## Checklist

- [ ] If any new components were added, there are exported from `packages/index.ts`
- [ ] This PR is associated with a JIRA and mentions in the commit message footer ("Closes …")
- [ ] This PR contains breaking changes and states in the commit message body ("BREAKING CHANGE: …")
- [x] I have reviewed the changes and provided detail to the sections above

### CODE REVIEW

File: packages/toggleWrapper/components/ToggleWrapper.tsx [hunk 1]
```diff
@@ -34,80 +34,61 @@ interface LocalToggleWrapperProps extends ToggleWrapperProps {
...
+    /* eslint-enable jsx-a11y/role-supports-aria-props */
+  );
+};
+
+ToggleWrapper.defaultProps = {
+  type: "checkbox",
+  value: ""
+};
 
 export default ToggleWrapper;
```
Comment (nataliepina) [inline packages/toggleWrapper/components/ToggleWrapper.tsx:94]: To retain the functionality of a PureComponent, could you memoize `ToggleWrapper`? We have a similar implementation throughout the codebase.

`export default React.memo(ToggleWrapper);`

File: packages/toggleWrapper/components/ToggleWrapper.tsx [hunk 2]
```diff
@@ -34,80 +34,61 @@ interface LocalToggleWrapperProps extends ToggleWrapperProps {
...
+      />
+      {children({ isActive, hasFocus })}
+    </label>
+    /* eslint-enable jsx-a11y/role-supports-aria-props */
+  );
+};
+
+ToggleWrapper.defaultProps = {
+  type: "checkbox",
+  value: ""
```
Comment (nataliepina) [inline packages/toggleWrapper/components/ToggleWrapper.tsx:91]: Could you bump up these two defaults as fallbacks in the set of props being passed in and destructured at the top?

File: packages/toggleWrapper/components/ToggleWrapper.tsx [hunk 3]
```diff
@@ -34,80 +34,61 @@ interface LocalToggleWrapperProps extends ToggleWrapperProps {
...
-    );
-  }
+    if (onFocus) {
+      onFocus(e);
+    }
+  };
 
-  private handleFocus(e) {
-    this.setState({ hasFocus: true });
+  const handleBlur = (e: React.FocusEvent<HTMLInputElement>) => {
```
Comment (nataliepina) [inline packages/toggleWrapper/components/ToggleWrapper.tsx:56]: Great work improving the types throughout 👍🏻

**Review by seialkali — APPROVED** [review]
Looks good to me! Thanks so much for your work on this!

**Review by nataliepina — APPROVED** [review]
Another great refactor, thanks @leonardomaier 👍🏻

**Pull request merged**